ARM-wvd-templates/HCI/AddHciVirtualMachinesTemplate.json

```diff
@@ -0,0 +1,323 @@
+        "type": "string",
+        "metadata": {
+          "description": "A username to be used as the virtual machine administrator account. The vmAdministratorAccountUsername and  vmAdministratorAccountPassword parameters must both be provided. Otherwise, domain administrator credentials provided by domainAdministratorUsername and domainAdministratorPassword will be used."
+        },
+        "defaultValue": ""
+      },
+      "vmAdministratorAccountPassword": {
+        "type": "securestring",
+        "metadata": {
+          "description": "The password associated with the virtual machine administrator account. The vmAdministratorAccountUsername and  vmAdministratorAccountPassword parameters must both be provided. Otherwise, domain administrator credentials provided by domainAdministratorUsername and domainAdministratorPassword will be used."
+        },
+        "defaultValue": ""
+      },
+      "oUPath": {
+        "type": "string",
+        "metadata": {
+          "description": "OU Path for the domain join"
+        },
+        "defaultValue": ""
+      },
+      "domain": {
+        "type": "string",
+        "metadata": {
+          "description": "Domain to join"
+        },
+        "defaultValue": ""
+      },
+      "domainAdministratorUsername": {
+        "type": "string",
+        "metadata": {
+          "description": "A username in the domain that has privileges to join the session hosts to the domain. For example, 'vmjoiner@contoso.com'."
+        },
+        "defaultValue": ""
+      },
+      "domainAdministratorPassword": {
+        "type": "securestring",
+        "metadata": {
+          "description": "The password that corresponds to the existing domain username."
+        },
+        "defaultValue": ""
+      },
+      "vmResourceGroup": {
+        "type": "string",
+        "metadata": {
+          "description": "The resource group of the session host VMs."
+        }
+      },
+      "vmLocation": {
+        "type": "string",
+        "metadata": {
+          "description": "The location of the session host VMs."
+        }
+      },
+      "customLocationId": {
+        "type": "string",
+        "metadata": {
+          "description": "A deployment target created and customized by your organization for creating virtual machines. The custom location is associated to an Azure Stack HCI cluster."
+        }
+      },
+      "virtualProcessorCount": {
+        "type": "int",
+        "metadata": {
+          "description": "Virtual Processor Count"
+        },
+        "defaultValue": 4
+      },
+      "memoryGB": {
+        "type": "int",
+        "metadata": {
+          "description": "Memory in GB."
+        },
+        "defaultValue": 8
+      },
+      "maximumMemoryGB": {
+        "type": "int",
+        "metadata": {
+          "description": "This paramater is optional and only used if dynamicMemory = true. When using dynamic memory this setting is the maximum GB given to the VM."
```

Comment:
super small nit, spelling.  paramater vs. parameter.  
there are a couple of these in the file below also.